akka-cluster/src/main/scala/akka/cluster/MembershipState.scala

```diff
@@ -119,4 +124,137 @@ import akka.annotation.InternalApi
       .map(_.uniqueAddress)
   }
 
+  def validNodeForGossip(node: UniqueAddress): Boolean =
+    node != selfUniqueAddress && isReachableExcludingDownedObservers(node)
```

Comment:
> failure detection should happen from the same N

right, that is my thinking

> limiting to the N first gossipable nodes 

The problem with that is that when there is a real network partition we will eventually try with all.